From 83af14cdbfe2eb1fbc5ce8ea395244dbe65dd4f3 Mon Sep 17 00:00:00 2001
From: Ralf Wildenhues <Ralf.Wildenhues@gmx.de>
Date: Wed, 9 Dec 2009 07:20:16 +0100
Subject: [PATCH 2/2] Fix 2.64 AC_TYPE_INT*_T macro body text regression.

* lib/autoconf/types.m4 (_AC_TYPE_INT_BODY): Move helper enum
definition to prologue section, to avoid syntax error.
* NEWS, THANKS: Update.
Report by Pierre Ynard.

Signed-off-by: Ralf Wildenhues <Ralf.Wildenhues@gmx.de>
---
 ChangeLog             |    8 ++++++++
 NEWS                  |    3 +++
 THANKS                |    1 +
 lib/autoconf/types.m4 |   12 ++++++------
 4 files changed, 18 insertions(+), 6 deletions(-)

diff --git ChangeLog ChangeLog
index 521ef79..b254d5e 100644
--- ChangeLog
+++ ChangeLog
@@ -1,3 +1,11 @@
+2009-12-09  Ralf Wildenhues  <Ralf.Wildenhues@gmx.de>
+
+	Fix 2.65 AC_TYPE_INT*_T macro body text regression.
+	* lib/autoconf/types.m4 (_AC_TYPE_INT_BODY): Move helper enum
+	definition to prologue section, to avoid syntax error.
+	* NEWS, THANKS: Update.
+	Report by Pierre Ynard.
+
 2009-11-24  Ralf Wildenhues  <Ralf.Wildenhues@gmx.de>
 
 	Fix AC_FUNC_MMAP regression with C++ compiler in 2.65.
diff --git NEWS NEWS
index d05b78c..436d22e 100644
--- NEWS
+++ NEWS
@@ -1,5 +1,8 @@
 GNU Autoconf NEWS - User visible changes.
 
+** The macros AC_TYPE_INT8_T, AC_TYPE_INT16_T, AC_TYPE_INT32_T, and
+   AC_TYPE_INT64_T work again.  Regression introduced in 2.65.
+
 ** AC_FUNC_MMAP works in C++ mode again.  Regression introduced in 2.65.
 
 * Major changes in Autoconf 2.65 (2009-11-21) [stable]
diff --git THANKS THANKS
index fdd6930..5ec7633 100644
--- THANKS
+++ THANKS
@@ -307,6 +307,7 @@ Peter Stephenson            pws@csr.com
 Philipp Thomas              kthomas@gwdg.de
 Philippe De Muyter          ?
 Pierre                      pierre42d@9online.fr
+Pierre Ynard                linkfanel@yahoo.fr
 Pontus Skoeld               pont@soua.net
 Rainer Orth                 ro@TechFak.Uni-Bielefeld.DE
 Raja R Harinath             harinath@cs.umn.edu
diff --git lib/autoconf/types.m4 lib/autoconf/types.m4
index 7a73fc2..3829c4c 100644
--- lib/autoconf/types.m4
+++ lib/autoconf/types.m4
@@ -635,14 +635,14 @@ m4_define([_AC_TYPE_INT_BODY],
 	 'long long int' 'short int' 'signed char'; do
        AC_COMPILE_IFELSE(
 	 [AC_LANG_BOOL_COMPILE_TRY(
-	    [AC_INCLUDES_DEFAULT],
-	    [enum { N = $[]2 / 2 - 1 };
-	     0 < ($ac_type) ((((($ac_type) 1 << N) << N) - 1) * 2 + 1)])],
+	    [AC_INCLUDES_DEFAULT
+	     enum { N = $[]2 / 2 - 1 };],
+	    [0 < ($ac_type) ((((($ac_type) 1 << N) << N) - 1) * 2 + 1)])],
 	 [AC_COMPILE_IFELSE(
 	    [AC_LANG_BOOL_COMPILE_TRY(
-	       [AC_INCLUDES_DEFAULT],
-	       [enum { N = $[]2 / 2 - 1 };
-		($ac_type) ((((($ac_type) 1 << N) << N) - 1) * 2 + 1)
+	       [AC_INCLUDES_DEFAULT
+	        enum { N = $[]2 / 2 - 1 };],
+	       [($ac_type) ((((($ac_type) 1 << N) << N) - 1) * 2 + 1)
 		 < ($ac_type) ((((($ac_type) 1 << N) << N) - 1) * 2 + 2)])],
 	    [],
 	    [AS_CASE([$ac_type], [int$[]2_t],
-- 
1.6.5.3

